javatests/arcs/core/data/expression/ParserTest.kt

```diff
@@ -92,22 +92,43 @@ class ParserTest {
     }
 
     @Test
-    fun parseScopeLookup() {
-        val fieldExpr = PaxelParser.parse("x")
-        assertThat(fieldExpr).isInstanceOf(Expression.FieldExpression::class.java)
-        fieldExpr as Expression.FieldExpression<Any>
-        assertThat(fieldExpr.qualifier).isNull()
-        assertThat(fieldExpr.field).isEqualTo("x")
-
-        val fieldExpr2 = PaxelParser.parse("x.y")
-        assertThat(fieldExpr2).isInstanceOf(Expression.FieldExpression::class.java)
-        fieldExpr2 as Expression.FieldExpression<Any>
-        assertThat(fieldExpr2.qualifier).isNotNull()
-        assertThat(fieldExpr2.field).isEqualTo("y")
-
-        val qualifier = fieldExpr2.qualifier as Expression.FieldExpression<Any>
+    fun parseScopeLookup_noQualifier() {
+        val expr = PaxelParser.parse("x")
+        assertThat(expr).isInstanceOf(Expression.FieldExpression::class.java)
+        expr as Expression.FieldExpression<Any>
+        assertThat(expr.qualifier).isNull()
+        assertThat(expr.field).isEqualTo("x")
+        assertThat(expr.nullSafe).isEqualTo(false)
+    }
+
+    @Test
+    fun parseScopeLookup_qualifier() {
+        val expr = PaxelParser.parse("x.y")
+        assertThat(expr).isInstanceOf(Expression.FieldExpression::class.java)
+        expr as Expression.FieldExpression<Any>
+        assertThat(expr.qualifier).isNotNull()
+        assertThat(expr.field).isEqualTo("y")
+        assertThat(expr.nullSafe).isEqualTo(false)
+
+        val qualifier = expr.qualifier as Expression.FieldExpression<Any>
+        assertThat(qualifier.qualifier).isNull()
+        assertThat(qualifier.field).isEqualTo("x")
+        assertThat(qualifier.nullSafe).isEqualTo(false)
+    }
+
+    @Test
+    fun parseScopeLookup_qualifier_nullSafe() {
+        val expr = PaxelParser.parse("x?.y")
```

Comment:
add a test case with a query, e.g. parse("x?.y + ?foo.bar")

Reply:
`?foo.bar` is not currently supported by the parser. Arguments cannot be qualifiers for scope lookups.